core/app/models/spree/line_item/pricers/abstract.rb

```diff
@@ -0,0 +1,42 @@
+module Spree
+  class LineItem
+    # Pricers for line items expose `#price`, returning a `Spree::Money` or `Money` object,
+    # and are used to choose or calculate the correct price for a line item.
+    # Ideally, they are idempotent, that is: When run on the same line item, they will return
+    # the same price.
+    #
+    module Pricers
+      # The `Abstract` pricer is here for other pricers to inherit from. It defines the common
+      # interface for pricers, as well as two practical errors, `VariantMissing` and
+      # `CurrencyMissing`.
+      #
+      # @attr [Spree::LineItem] The line item to calculate or choose a price for
+      class Abstract
+        class VariantMissing < StandardError; end
+        class CurrencyMissing < StandardError; end
+
+        attr_reader :line_item
+
+        def initialize(line_item)
+          @line_item = line_item
+        end
+
+        # Returns the calculated or chosen price
+        #
+        # @return [Spree::Money, Money] an money object representing line item's price
```

Comment:
A pricer's return value will be sent `to_d` and `currency`, expecting `::Money`s signature. I thought I'd allow a plain money object to start phasing out `Spree::Money`.